### PR TITLE
Typing for tests 1/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

add typing for test parameterization so it is easier to read and validate paraeters

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
I don't see the point of annotating tests, we will never check them with mypy and if they didn't work they would fail...

Our efforts would be better spent annotating the codebase or fixing stuff

**Comment by Borda** [comment]
> I don't see the point of annotating tests, we will never check them with mypy and if they didn't work they would fail...
> 
> Our efforts would be better spent annotating the codebase or fixing stuff

ok, I agree that there is no point in adding more typing, just found it easier to read the test, and anyway we need to refactor our testing...

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/core/test_memory.py [hunk 1]
```diff
@@ -97,10 +97,7 @@ def test_invalid_weights_summmary():
         Trainer(weights_summary='temp')
 
 
-@pytest.mark.parametrize(['mode'], [
-    pytest.param(ModelSummary.MODE_FULL),
-    pytest.param(ModelSummary.MODE_TOP),
-])
+@pytest.mark.parametrize('mode', [ModelSummary.MODE_FULL, ModelSummary.MODE_TOP])
 def test_empty_model_summary_shapes(mode):
```
Comment (akihironitta) [inline tests/core/test_memory.py:101]: Not sure if we need typing here, but no typing for mode?
```suggestion
def test_empty_model_summary_shapes(mode: ModelSummary):
```

**Review by akihironitta — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]
